src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -0,0 +1,207 @@
+		},
+	}
+	cmd.Flags().StringArrayVarP(&partitions, "partition", "p", nil, "Topic-partitions to move and new replica locations (repeatable)")
+	cmd.MarkFlagRequired("partition")
+	return cmd
+}
+
+// extractNTP parses the partition flag with format; foo/0:1,2,3 or 0:1,2,3.
+// It extracts letters before the colon and formats it.
+func extractNTP(topic string, ntp string) (ns string, tp string, part int, err error) {
+	re := regexp.MustCompile(`^((?:[^:]+/)?\d+):.*$`)
+	m := re.FindStringSubmatch(ntp)
+	if len(m) == 0 {
+		return "", "", -1, fmt.Errorf("invalid format for %s", ntp)
+	}
+	beforeColon := m[1]
+	if topic != "" {
+		part, err = strconv.Atoi(beforeColon)
+		if err != nil {
+			return "", "", -1, fmt.Errorf("%s", err)
+		}
+	} else if n := strings.Split(beforeColon, "/"); len(n) == 3 {
+		ns = n[0]
+		tp = n[1]
+		part, err = strconv.Atoi(n[2])
+		if err != nil {
+			return "", "", -1, fmt.Errorf("%s", err)
+		}
+	} else if len(n) == 2 {
+		ns = "kafka"
+		tp = n[0]
+		part, err = strconv.Atoi(n[1])
+		if err != nil {
+			return "", "", -1, fmt.Errorf("%s", err)
+		}
+	} else {
+		return "", "", -1, fmt.Errorf("invalid format for %s", ntp)
+	}
+	return ns, tp, part, nil
+}
+
+// configureReplicas parses the partition flag with format; foo/0:1-0,2-1,3-2 or 0:1,2,3
+// It extracts letters after the colon and return as a slice of []adminapi.Replica.
+func configureReplicas(cmd context.Context, cl *adminapi.AdminAPI, partition string, currentReplicas []adminapi.Replica) ([]adminapi.Replica, error) {
+	re := regexp.MustCompile(`^[^:]+:(\d+(?:-\d+)?(?:,\d+(?:-\d+)?)*)$`)
+	m := re.FindStringSubmatch(partition)
+	if len(m) == 0 {
+		return nil, fmt.Errorf("invalid format for %s", partition)
+	}
+	var newReplicas []adminapi.Replica
+	for _, nodeCore := range strings.Split(m[1], ",") {
+		if split := strings.Split(nodeCore, "-"); len(split) == 1 {
+			node, _ := strconv.Atoi(nodeCore)
+			core := findCore(cmd, cl, node, currentReplicas)
+			newReplicas = append(newReplicas, adminapi.Replica{
+				NodeID: node,
+				Core:   core,
+			})
+		} else {
+			node, _ := strconv.Atoi(split[0])
+			core, _ := strconv.Atoi(split[1])
+			newReplicas = append(newReplicas, adminapi.Replica{
+				NodeID: node,
+				Core:   core,
+			})
+		}
+	}
+	return newReplicas, nil
+}
+
+// findCore finds a shard (CPU core) where a replica should be assigned on.
+// If 'core' is specified, use it.
+// If 'core' is not specified for a new node, find a shard in a random-fashion.
+// If 'core' is not specified for an existing node, use the current shard as is.
+func findCore(cmd context.Context, cl *adminapi.AdminAPI, nodeID int, currentReplicas []adminapi.Replica) int {
+	for _, i := range currentReplicas {
+		if nodeID == i.NodeID {
+			return i.Core
+		}
+	}
+	broker, _ := cl.Broker(cmd, nodeID)
```

Comment:
This right here will cause the Broker request to run semi-serially -- each replica with a 0 core runs this concurrently, but it's possible for this to run tons of times for the same broker. I think what this should change to: after configureReplicas above, for all partitions are moving to a new node, _then_ request each of those node's Broker information (requesting this concurrently), and then finally map those unmapped replicas to specific cores.

Basically the aim with this comment is to make roundtrips to the cluster as minimal as possible and as concurrent as possible.

Reply:
That's indeed a good point to make the command reliable with a lesser number of admin api calls. 
Let me confirm if I'm understanding what you meant correctly; first we should finish the `for _, partition := range partitions {` loop in the main Run function without calling `cl.MoveReplicas` because we don't yet decide a core to be assigned for new nodes. In a new for-loop, running concurrently, then we call `cl.Broker` to decide a core for new nodes. Lastly, we run another for-loop, being executed concurrently, and call `cl.MoveReplicas` finally. 

Am I getting you?

Reply:
Yes I think that'd do it -- and in the ideal case (no Broker call needed), the flow you mentioned^^ will be ~nearly exactly the same as right now.

If you want to get into optimizing this further -- `MoveReplicas` can be called concurrent with `Broker` for any partition where we already know the move destination. We don't _need_ to strictly serialize `parse -> broker -> move`, it can be `parse -> (broker -> move | move)`

Reply:
Pushed a new commit, https://github.com/redpanda-data/redpanda/pull/13684/commits/ce71d15f0ab09aa087b3d5d4a1fe7554f65fe724. 

I kinda feel this part `(broker -> move | move)` should be serial in order to refer to the updated core count for a node which was already requested. Thoughts?

Reply:
Another reasons on not making `(broker -> move | move)` concurrent are; 

* making everything fail when one of the requests fails doesn't make sense. A failure of a request shouldn't affect the other request as they're independent.
* capturing which request succeeds or fails correctly and reporting it to the user at the end.